src/cluster.c

```diff
@@ -3872,6 +3872,24 @@ void clusterLogCantFailover(int reason) {
     serverLog(LL_WARNING,"Currently unable to failover: %s", msg);
 }
 
+/* This function emits a log when an election is attempted but doesn't succeed because the quorum is not reached. */
+static void clusterLogNeededquorumNotReached(int cur_vote, int cur_quorum) {
```

Comment:
Why is this a separate function compared to just adding the logic to clusterLogCantFailover? Seems like we are duplicating the logic for deduplicating failover logs.

Reply:
In clusterLogCantFailover, the message is created with char*, appending integers to char* will need to create a char[] points to this pointer first then use snprintf() for appending, also we do not want to print the log again if we have the same vote condition as previous one. The amount of coding will break the consistency with the other reason, so I think it's better to have another function to write for it.

Reply:
I would rather have one coherent function than two separate functions. It's pretty easy to just add the serverLog() you added in this function inside the switch statement.

Reply:
But `clusterLogCantFailover()` is called for other reasons than a quorum not reached. So this log does not make sense in the other cases. We could pass 0 as the `cur_vote` and `cur_quorum` in those cases and add some logic to not log the quorum message in this case, but that seems a bit convoluted.

Reply:
I'm not really convinced we don't *always* want to know how many votes we had. This is during an election that failed, so it would be good to know the progress, I think all places we call clusterLogCantFailover have the vote information.

Reply:
Another thing is, if we just add the serverLog() into `clusterLogCantFailover()`, the log order would be messed. It would first print out the serverLog thing, then log `can not fail over`, lastly the reason. Unless we added another if logic again after msg logged, which would break the consistency. Thus we use another function that is called after `clusterLogCantFailover()` finished.

Reply:
I don't think the consistency matters here.